yaml_requests/utils/template.py

```diff
@@ -30,37 +34,35 @@ def _contains_template(self, str_in):
         has_end = self.variable_end_string in str_in
         return has_start and has_end
 
-    def _is_template(self, str_in):
-        start_eq = str_in.startswith(self.variable_start_string)
-        end_eq = str_in.endswith(self.variable_end_string)
-        single_start_eq = str_in.count(self.variable_start_string) == 1
-        return start_eq and single_start_eq and end_eq
-
-    def _resolve_string(self, str_in):
+    def _resolve_string(self, str_in) -> any:
         if not self._contains_template(str_in):
             return str_in
 
-        if self._is_template(str_in):
-            str_in = str_in.replace('}}', ' | to_json }}')
         template = self.from_string(str_in)
         rendered = template.render()
 
-        if 'to_json' in str_in:
+        try:
             try:
-                return json.loads(rendered)
-            except Exception:
-                pass
-
-        return rendered
-
-    def _resolve_dict(self, item):
+                self._json_decoder.scan_once(rendered, 0)
+            except json.JSONDecodeError:
+                # it's yaml
+                is_yaml = True
+            except StopIteration:
+                # it's not JSON
+                return rendered
+
+            return yaml.load(rendered, Loader=yaml.SafeLoader)
+        except Exception:
+            pass
```

Comment:
Should this also be `return rendered`? I guess the execution would end up here if both JSON and YAML load fail, but not sure when that would happen 🤔